### ISSUE TITLE
Status of new color support?

### ISSUE THREAD

**Issue by shmerl** [dirruk1/gnome-breeze#74]
A while ago there was an effort to refactor breeze-gtk to have one theme that generates style assets based on color settings changes in KDE (using a script). Did this move anywhere? I see that KDE Plasma 5.10.5 is still bundling old static themes with it.

**Comment by ghost** [comment]
I'm sorry to say things haven't moved much in this area. It's still nowhere near as straightforward as it was with theme-engine support, and there are more implementation details that need to be taken care of outside of the theme. But, aside from that, we've both been busy with other stuff.

I'm personally a bit burnt out on the work since there are so many barriers to getting a good experience from the GTK side of things, despite attempts to cooperate or find some workaround for CSDs, dynamic theme loading, and more. I know this may be a bit personal and unprofessional, but I've been trying to do something to keep the default themes compatible across toolkits/DEs for a while now (starting with Adwaita GTK 2), and I've gotten a bit sick of the overwhelmingly negligent attitude outside of KDE. I'd rather work on projects where I feel I'm making a real difference instead of fighting a losing battle.

Of course, dirruk1 has put a lot more work into this theme than I have- I've just been trying to coordinate things on the higher level to no avail. So really, we should give him all the support he needs to continue on and see if any other experienced themers are willing to pitch in. While things may never be as good as they were before, I think there's still room for the theme to grow and push against those boundaries.

**Comment by shmerl** [comment]
Sorry to ping about it again. How are things going with this? May be disregaridng GTK2 (which is a different beast to support), can at least GTK3 colors management be brushed up? GTK2 is becoming increasingly rare now, with only some holdouts remaining (like Gimp), but going forward it should be less and less of a problem.

**Comment by alex65536** [comment]
@shmerl, it there's still no straightforward way to do it. But there's a workaround: you can rebuild the theme each time you change the color scheme. You can clone the [official repo](https://github.com/KDE/breeze-gtk) (it has changed much recently), and use `cd src && ./build_theme.sh`, which will use your current color scheme to rebuild the theme and install it into `~/.local/share/themes`. Of course, you will need to install the dependencies, such as `sassc` and `python3-cairo`.

**Comment by alex65536** [comment]
AFAIK, Breeze GTK in Plasma 5.17 will support color schemes: https://phabricator.kde.org/D22844.

**Comment by shmerl** [comment]
Is it a merge of those scripts which generate GTK assets from KDE color scheme?

**Comment by alex65536** [comment]
@shmerl, no. They are now using GTK's named colors. So, overriding the colors in `~/.config/gtk-3.0/gtk.css` would change the colors in the theme. This prevents the theme from rebuilding on user's machine. Colors KCM is able to regenerate color definitions.

**Comment by shmerl** [comment]
Interesting, thanks!
